packages/core/status/core-status-server-internal/src/routes/status.ts

```diff
@@ -88,60 +92,103 @@ export const registerStatusRoute = ({
       },
     },
     async (context, req, res) => {
-      const { version, buildSha, buildNum, buildDate } = config.packageInfo;
-      const versionWithoutSnapshot = version.replace(SNAPSHOT_POSTFIX, '');
+      const authRequired = !config.allowAnonymous;
+      const isAuthenticated = req.auth.isAuthenticated;
+      const redactedStatus = authRequired && !isAuthenticated;
       const [overall, coreOverall, core, plugins] = await firstValueFrom(combinedStatus$);
 
-      const { v8format = true, v7format = false } = req.query ?? {};
-
-      let statusInfo: StatusInfo | LegacyStatusInfo;
-      if (!v7format && v8format) {
-        statusInfo = {
-          overall,
-          core,
-          plugins,
-        };
-      } else {
-        incrementUsageCounter({ counterName: 'status_v7format' });
-        statusInfo = calculateLegacyStatus({
-          overall,
-          core,
-          plugins,
-          versionWithoutSnapshot,
-        });
-      }
-
-      const lastMetrics = await firstValueFrom(metrics.getOpsMetrics$());
-
-      const body: StatusHttpBody = {
-        name: config.serverName,
-        uuid: config.uuid,
-        version: {
-          number: versionWithoutSnapshot,
-          build_hash: buildSha,
-          build_number: buildNum,
-          build_snapshot: SNAPSHOT_POSTFIX.test(version),
-          build_date: buildDate.toISOString(),
-        },
-        status: statusInfo,
-        metrics: {
-          last_updated: lastMetrics.collected_at.toISOString(),
-          collection_interval_in_millis: metrics.collectionInterval,
-          os: lastMetrics.os,
-          process: lastMetrics.process,
-          processes: lastMetrics.processes,
-          response_times: lastMetrics.response_times,
-          concurrent_connections: lastMetrics.concurrent_connections,
-          requests: {
-            ...lastMetrics.requests,
-            status_codes: lastMetrics.requests.statusCodes,
-          },
-          elasticsearch_client: lastMetrics.elasticsearch_client,
-        },
-      };
+      const responseBody = redactedStatus
+        ? getRedactedStatusResponse({ coreOverall })
+        : await getFullStatusResponse({
+            incrementUsageCounter,
+            config,
+            query: req.query,
+            metrics,
+            statuses: { overall, core, plugins },
+          });
```

Comment:
One of the root change of the PR:
- Access of the `/api/status` route switch to `authRequired: 'optional'`
- If auth is required (based on `config.allowAnonymous`) and the request was not authenticated, then we return a redacted body instead of the full payload.